src/transformed_distribution.jl

```diff
@@ -1,20 +1,19 @@
+function variateform(d::Distribution, b)
```

Comment:
Do we need this function? AFAICT it's only used in the constructor of `TransformedDistribution` below. Could we directly use `Distributions.variate_form` instead?

Reply:
> Do we need this function? AFAICT it's only used in the constructor of TransformedDistribution below.

Yeah I introduced because of the constructor since we need to determine what the variate form is going to be using the underlying distribution and the transformation. 

> Could we directly use Distributions.variate_form instead?

I don't quite see how that method helps. AFAIK that is only meant to return the variateform of a given `Sampleable` type; here we're just trying to determine what the variate form should be when we combine a distribution with a transformation.

Reply:
Yeah, I guess you're right - I assumed one could use something like `variate_form(b(d))` but the point is just to compute the variate type of "`b(d)`" here.